src/main/java/com/github/webicitybrowser/webicitybrowser/engine/protocol/WebicityProtocol.java

```diff
@@ -0,0 +1,49 @@
+package com.github.webicitybrowser.webicitybrowser.engine.protocol;
+
+import com.github.webicitybrowser.spec.url.URL;
+import com.github.webicitybrowser.webicity.core.net.Connection;
+import com.github.webicitybrowser.webicity.core.net.Protocol;
+import com.github.webicitybrowser.webicity.core.net.ProtocolContext;
+
+import java.io.*;
+
+public class WebicityProtocol implements Protocol {
+	@Override
+	public String[] getSchemes() {
+		return new String[]{"webicity"};
+	}
+
+	@Override
+	public Connection openConnection(URL url, ProtocolContext context) throws IOException {
+
+		StringBuilder path = new StringBuilder(
+			url.toString().substring( url.getHost().length() + 3)
```

Comment:
It seems like the path actually gets treated like a host in this case. So you can just do url.getHost() for before the slash
If you would like any path after the slash, url.getHost() + "/" + url.getPath() should work.

I would not recommend converting the URL to a string and slicing it like this, as that is error prone

Reply:
i saw that when there is only file name the "/" should be removed but when there is path like "src/index" doing as you said works fine. I committed changes and they are in this pull request i suppose.